packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -117,6 +119,10 @@ export const getLocationSearch = (req, opts = {}) => {
  * @return {Promise}
  */
 export const render = async (req, res, next) => {
+    const includeServerTimingHeader = '__server_timing' in req.query
+    const shouldTrackPerformance = includeServerTimingHeader || process.env.SERVER_TIMING
+    res.__performanceTimer = new PerformanceTimer({enabled: shouldTrackPerformance})
```

Comment:
Have you thought about placing the `performanceTimer` at `res.locals.performanceTimer`? ... not sure about this, but it seems to be the purpose of that locals object.

Reply:
Yea, I intentionally don't want to add it to `res.locals` because this is a grey area in our PWA Kit implementation, since `res.locals` is used and shared in the userland. The locals object is being passed to AppConfig component as a prop and also in the `restore` / `extraGetPropsArgs` methods. 

I don't want developer making mistakes cause the app to serialize the performance timer, since the performance timer is a js class instance, it's not serializable.

A bad example that i want to avoid:

```
AppConfig.extraGetPropsArgs = (locals = {}) => {
    // do random things here
    // returned locals
    return {
        ...locals
    }
}
```

Reply:
I can get on board with that, better to be save that sorry. I know that the locals object was kinda of regret that Oli had as it was constantly mis-used.